### PR TITLE
Fix CORS and mixed content issue

### PR DESCRIPTION
**Pull request by hosseinnedaee**
In this PR I used  Nuxtjs serverMiddleware to create a proxy server and pass the requests through the proxy and get over CORS and mixed-content issue's

resolving https://github.com/liyasthomas/postwoman/issues/175 and https://github.com/liyasthomas/postwoman/issues/2

### CODE REVIEW

File: pages/index.vue [hunk 1]
```diff
@@ -821,13 +838,24 @@
...
             this.$refs.historyComponent.addEntry(entry);
             return;
+          } else {
+            this.response.status = error.message;
+            this.response.body = "See JavaScript console (F12) for details.";
+            this.$toast.error('Something went wrong!', {
+              icon: 'error'
+            });
+            if(!this.$store.state.postwoman.settings.PROXY_ENABLED) {
+              this.$toast.info('Turn on the proxy', {
```
Comment (NBTX) [inline pages/index.vue]: This is looking good to me, my only recommendation is that this be renamed to 'Try enabling the proxy' or words to that effect, but it's not that significant.

Comment (liyasthomas) [reply pages/index.vue]: How about "Switch to Proxy mode"?

**Comment by NBTX** [comment]
Oh- just one other issue, we need to add `Access-Control-Allow-Headers: *` to the proxy response

File: pages/index.vue (hunk 1 above)
Comment (NBTX) [reply pages/index.vue]: Yes - but perhaps with the question mark. The idea being to imply to the user that it's something that might help with the error.

**Comment by NBTX** [comment]
See #200 which includes these commits with some finishing touches and bug fixes.

**Pull request merged**